classes/WorldGenerator.py

```diff
@@ -347,33 +350,33 @@ def noise3(self, x, y, z):
 
 
 class WorldGenerator(SimplexNoise):
-    
-    async def generateNewChunk(self, x, y, z, width, height, region) -> Chunk:
+
+    async def generateNewChunk(self, x, y, z, width, height, chunk) -> Chunk:
         positions = []
         for blockX in range(1, width):
             for blockZ in range(1, width):
                 bY = self.noise2(blockX, blockZ)  # Use 2D noise to generate the y coordinate, then use 3D noise to
                 # generate everything else
                 blockY = scaleNoise(bY, (63, 80))  # Scale the noise to be between min-max y value
                 positions.append((blockX, blockY, blockZ))
-        chunk = Chunk(x, y, z, [], region, width, height)
-        await self._regenerate_chunk(x, y, z, region, positions, chunk)
+        await self._regenerate_chunk(x, y, z, positions, chunk)
         return chunk
 
-    async def _regenerate_chunk(self, x, y, z, region, positions, chunk: Chunk):
+    async def _regenerate_chunk(self, x, y, z, positions, chunk: Chunk):
+        # WTF ?
```

Comment:
If I remember it correctly I did that to make the code run faster as you shouldn't confuse ChunkColum (which will be added later) and Chunk (also known as ChunkSection), this regenerates a single 16 x 16 x 16 Chunk section, which results in the fact that positions needs to be regenerated pretty often. Now I reckon that you could simply limit the size of the positions array, but If I recall correct there was another limitation with (tree?) generation that required the full positions array to be known.

Maybe the generators should just be generating everything after the initial generation, I suppose that that would actually be more efficent in terms of inter-chunk structure generation or this one

Reply:
The vanilla game works by first generating the world's overall terrain and shape (mesa goes here, mountain is at y142, etc). Then it adds basic world features like caves, rivers, oceans, water lakes, etc. and then does more complex non-natural features like villages, strongholds, and the like

Reply:
That's what I have in mind, which we would need to do later no

Reply:
I've deleted the commit that has changed this class, I'll edit it later